ext/Shared/__init__.lua

```diff
@@ -263,124 +51,129 @@ local function GetCustomLevel(p_LevelName, p_GameModeName)
 	return s_Preset
 end
 
+Hooks:Install('ResourceManager:LoadBundles', 100, function(p_Hook, p_Bundles, p_Compartment)
+	print(p_Compartment)
+	print(p_Bundles[1])
+end)
+
 -- nº 1 in calling order
-Events:Subscribe('Level:LoadResources', function()
+Events:Subscribe('Level:LoadResources', function(p_LevelName, p_GameMode, p_IsDedicatedServer)
 	print("-----Loading resources")
-	m_ObjectVariations = {}
-	m_PendingVariations = {}
 
-	m_CustomLevelData = GetCustomLevel(SharedUtils:GetLevelName(), SharedUtils:GetCurrentGameMode())
-end)
+	local s_SuperBundleName = string.gsub(p_LevelName, 'Levels', BUNDLE_PREFIX)
+	print("MountSuperBundle: " .. s_SuperBundleName)
+	ResourceManager:MountSuperBundle(s_SuperBundleName)
 
--- nº 2 in calling order
-Events:Subscribe('Partition:Loaded', function(p_Partition)
-	if not m_CustomLevelData then
-		return
-	end
+	m_CustomLevelData = GetCustomLevel(p_LevelName, p_GameMode)
 
-	if p_Partition == nil then
-		return
-	end
+	for l_PartitionGuid, l_Instances in pairs(m_CustomLevelData) do
+		for _, l_InstanceGuid in ipairs(l_Instances) do
+			-- print(tostring(l_PartitionGuid) .. ' - ' .. tostring(l_InstanceGuid))
 
-	local s_PrimaryInstance = p_Partition.primaryInstance
+			ResourceManager:RegisterInstanceLoadHandlerOnce(Guid(l_PartitionGuid), Guid(l_InstanceGuid), function(p_Instance) 
+				p_Instance = _G[p_Instance.typeInfo.name](p_Instance)
+				p_Instance:MakeWritable()
+				p_Instance.excluded = true
+			end)
+		end
+	end
+end)
 
-	if s_PrimaryInstance == nil then
-		print('Instance is null? ' .. p_Partition.name, true)
+---Patches the level, adding a SubWorldReferenceObjectData to the level that references the SubWorld in the custom bundle
+local function _PatchLevel(p_LevelName)
+	local s_Data = LevelData(ResourceManager:SearchForDataContainer(SharedUtils:GetLevelName()))
+	s_Data:MakeWritable()
+
+	local s_SWROD = SubWorldReferenceObjectData(Guid('6a724d44-4efd-4f7e-9249-2230121d7ecc'))
+	
+	s_SWROD.bundleName = BUNDLE_PREFIX .. '/' .. p_LevelName:gsub(".*/", "") .. '/' .. SharedUtils:GetCurrentGameMode()
```

Comment:
Would be nice if this could be a config.
Like: `BundleName = string.format("%s/%s/%s", BUNDLE_PREFIX, LEVELNAME, GAMEMODE)`
So I would still be able to build the bundlename but change the order for example. Or just use the levelname and not the gamemode name.
Same for the superbundle name.